cocos/core/components/ui-base/canvas.ts

```diff
@@ -197,15 +198,40 @@ export class Canvas extends Component {
     }
 
     get visibility () {
-        if (this._camera) {
-            return this._camera.view.visibility;
+        if (this._cameraComponent) {
+            return this._cameraComponent.visibility;
         }
 
         return -1;
     }
 
+    @type(Camera)
+    @tooltip('2D渲染相机')
+    get cameraComponent () {
+        return this._cameraComponent;
+    }
+
+    set cameraComponent (value) {
+        if (this._cameraComponent === value) { return; }
+
+        this._cameraComponent = value;
+
```

Comment:
It seems that OnEnable is not executed when the camera component is dynamically set.

Reply:
Is it necessary?